backend-ktor/src/main/kotlin/me/zwsmith/plugins/recipes/RecipeController.kt

```diff
@@ -0,0 +1,158 @@
+package me.zwsmith.plugins.recipes
+
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.Application
+import io.ktor.server.request.receive
+import io.ktor.server.response.respond
+import io.ktor.server.routing.delete
+import io.ktor.server.routing.get
+import io.ktor.server.routing.post
+import io.ktor.server.routing.put
+import io.ktor.server.routing.routing
+import me.zwsmith.plugins.Ingredient
+import me.zwsmith.plugins.QuantifiedIngredient
+import me.zwsmith.plugins.database.RecipeService
+import org.jetbrains.exposed.sql.Database
+
+fun Application.recipeRouter() {
+    val database = Database.connect(
+        url = "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1",
+        user = "root",
+        driver = "org.h2.Driver",
+        password = "",
+    )
+    val recipeService = RecipeService(database)
```

Comment:
`recipeService` should probablu be passed into this function as a dependency rather than being created here.

Both the database connection, and the service should be instantiated in application/main when you're wiring up the dependencies for the application